### PR TITLE
fix buggy exp(-inf) on Windows

### PR DESCRIPTION
**Pull request by danmelamed**
https://github.com/JohnLangford/vowpal_wabbit/commit/35ccb4179582f69199702728bcb2d9fdf70e6919

### CODE REVIEW

**Comment by JohnLangford** [comment]
What is the dmtest stuff?

**Comment by JohnLangford** [comment]
Why are there changes to vw.sln?

**Comment by JohnLangford** [comment]
Also is this still somehow inducing segfaults?  I don't see anything odd, except that you might want to replace isinf() with infpattern().   I believe the latter is more cross-platform.

**Comment by danmelamed** [comment]
vw.sln changed to include the new header file.

dmtest was a separate commit that was not supposed to be part of this pull request; not sure why it was included.  Sorry, still figuring out the interface.  dmtest is just my own miniproject that I use for testing small code stubs.

**Comment by JohnLangford** [comment]
Generally, you want to make a branch with just the stuff for the pull request.

**Comment by JohnLangford** [comment]
I'd suggest editing vw.sln in a text editor---it seems that much more stuff is being changed than just adding a line for the header file.

**Comment by danmelamed** [comment]
isinf() is a standard function in C++11, and a standard macro in C99.  infpattern() seems to rely on particular encodings, which I imagine different platforms could implement differently....

**Comment by JohnLangford** [comment]
Either isinf() or isnan() turned out to not be cross-platform (I forget 
which).   Since this is only for windows, it doesn't matter much as long 
as it works on windows.

-John

**Comment by arielf** [comment]
I wonder if it is possible to leave `exp()` alone in all files where it appears, and (possibly) define it to whatever-else on windows only.  It seems cleaner than changing all `exp()` calls to an uglier and less intuitive/familiar form because of a specific bug on a specific platform.  Generally, we should try to leave simple/original call alone.  I once suggested `platform.h` for this.

See:  http://pastebin.com/thHtb7vr   (not tested, but someone has to start somewhere :sweat_smile: )

**Comment by JohnLangford** [comment]
I tried some #defines like this when I was first dealing with porting to
windows.   It did not seem to work well with visual studio unfortunately,
although I forget what the problem was exactly.

-John

File: vowpalwabbit/vw.sln [hunk 1]
```diff
@@ -63,6 +63,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		cs_unittest.testsettings = cs_unittest.testsettings
 	EndProjectSection
 EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "dmTest", "..\dmTest\dmTest.vcxproj", "{DFD56302-EC7E-4480-9A44-A0467887ED20}"
```
Comment (JohnLangford) [inline vowpalwabbit/vw.sln]: Can you remove these lines?

File: vowpalwabbit/vw.sln [hunk 2]
```diff
@@ -207,6 +209,17 @@ Global
 		{E4E962AE-7056-4EB0-A8C5-8DC824A4B068}.Release|Win32.ActiveCfg = Release|x64
 		{E4E962AE-7056-4EB0-A8C5-8DC824A4B068}.Release|x64.ActiveCfg = Release|x64
 		{E4E962AE-7056-4EB0-A8C5-8DC824A4B068}.Release|x64.Build.0 = Release|x64
+		{DFD56302-EC7E-4480-9A44-A0467887ED20}.Debug|Mixed Platforms.ActiveCfg = Debug|Win32
```
Comment (JohnLangford) [inline vowpalwabbit/vw.sln]: What does all of this do?

Comment (danmelamed) [inline vowpalwabbit/vw.sln]: It just sets x64 as the build target.  I thought I excluded my vw.sln from any commits.  Maybe I failed to include it when I reverted the dmTest project.  Sorry… will try again.

D

**Pull request merged**